eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -85,21 +87,58 @@ stages:
                 deploy:
                   steps:
                     - checkout: self
-                    - pwsh: |
-                        Get-ChildItem $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
-                        New-Item -Type Directory -Name ${{artifact.safeName}} -Path $(Pipeline.Workspace)
-                        Copy-Item $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.name}}-[0-9]*.[0-9]*.[0-9]* $(Pipeline.Workspace)/${{artifact.safeName}}
-                        Get-ChildItem $(Pipeline.Workspace)/${{artifact.safeName}}
-                      workingDirectory: $(Pipeline.Workspace)
+                    - template: /eng/pipelines/templates/steps/stage-filtered-artifacts.yml
+                      parameters:
+                        SourceFolder: ${{parameters.ArtifactName}} 
+                        TargetFolder: ${{artifact.safeName}}
+                        PackageName: ${{artifact.name}}
                       displayName: Stage artifacts
-                      timeoutInMinutes: 5
                     - template: /eng/common/pipelines/templates/steps/create-tags-and-git-release.yml
                       parameters:
                         ArtifactLocation: $(Pipeline.Workspace)/${{artifact.safeName}}
                         PackageRepository: Maven
                         ReleaseSha: $(Build.SourceVersion)
                         RepoId: Azure/azure-sdk-for-java
 
+
+          - ${{if ne(artifact.options.skipPublishDocs, 'true')}}:
+            - deployment: PublicDocsMs
+              displayName: "Publish Updates for Docs.MS"
+              condition: and(succeeded(), ne(variables['Skip.PublishDocs'], 'true'))
+              environment: github
+              timeoutInMinutes: 5
+              dependsOn: VerifyReleaseVersion
+
+              pool:
+                vmImage: vs2017-win2016
```

Comment:
orthogonal to your change but we probably should see about moving these to the latest windows image, windows-2019